vmdb/app/models/ems_kubernetes.rb

```diff
@@ -20,7 +27,8 @@ def self.raw_connect(hostname, port, api_version)
   def self.raw_api_endpoint(hostname, port)
     require 'uri'
 
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    port &&= port.to_i
+    uri = URI::HTTP.build(:path => "/api", :port => port)
```

Comment:
This was a bug...if the user did not enter a port, it should no be passed.  This code was formerly passing `0`